include/net/socket.h

```diff
@@ -41,6 +41,8 @@ struct zsock_pollfd {
 #define ZSOCK_POLLIN 1
 #define ZSOCK_POLLOUT 4
 #define ZSOCK_POLLERR 8
+#define ZSOCK_POLLHUP 0x10
```

Comment:
This looks like a bit field to me, why not use BIT() macro here for all of these defines?

Reply:
It is bitfield in the sense that values are ORed otgether. But the values are specific numerical values, afaik coming from iBCS, (https://en.wikipedia.org/wiki/Intel_Binary_Compatibility_Standard). I don't think it's good idea to "obfuscate" them using BIT(). (If it was just some local define without any external meaning assigned, there wouldn't be any objections.)

Reply:
This seems much easier to read and understand
```
#define ZSOCK_POLLIN BIT(0)
#define ZSOCK_POLLOUT BIT(2)
#define ZSOCK_POLLERR BIT(3)
#define ZSOCK_POLLHUP BIT(4)
```
etc

The actual values of these defines do not seem to matter according to http://pubs.opengroup.org/onlinepubs/009695399/basedefs/poll.h.html so there is no need to leave holes there.

Reply:
> The actual values of these defines do not seem to matter according to 

They don't, but we have a convention to follow values used in Linux, which in turn appears to follow iBCS.

Reply:
We do not need to copy the values from Linux in this case and using hex numbers just looks ugly here. Using BIT() macro makes lot of sense and is more readable.

Reply:
I'm sorry, but:

1. I'm the maintainer of this subsystem, so let me kindly say that we need values compatible with Linux here (just because we need them everywhere). If you'd like to challenge that, please open a ticket titled "Let's make Zephyr deliberately incompatible with existing well-known ABIs", and let's go thru the usual discussion, etc. process on it.
2. The new values added just follow the syntax used for values which are already there. Why, when in https://github.com/zephyrproject-rtos/zephyr/pull/10816#discussion_r227783233 , you're asked why you *don't* use *established* naming conventions, and you reply "because I just follow local naming of functions around" - the reply is "OK" (well, not sure about other reviewers, but I'm definitely ok with that, with a notice "let's fix it eventually"). Why do you keep pushing for the changes which has nothing to do with the matter of this PR?

Reply:
Are you really suggesting that applications would define POLLIN itself to use some arbitrary binary number from Linux? Of course applications would not do that, but they would use the value that the header file specifies. And as Posix does not specify that the POLLIN should have some specific value, there is no ABI issue here.

Reply:
No, I'm just telling that when a programmer prints flags values as a number (happens all the time), looks at them in the debugger, they will see the same values in Zephyr, Linux, and hopefully in other sane OSes. Helps avoid confusion greatly.

Note that I'm *not* saying about being able to run Linux/etc. binaries on Zephyr. However, some other people may. E.g. @lpereira talks about "when Zephyr will have processes", I know that dynaloading is in the planning, then some guy may make a nice weekend hack showing Zephyr running a simple ELF from Linux. And you want to kill all that fun! ;-)

Reply:
No need to argue over this thing, we can leave the numbers here if we want. What you describe about matching linux number and zephyr number does not make much sense imho, there are lot of number handling in linux and zephyr (for similar functionality/apis) and trying to match the values where it is not strictly needed is pointless.

Reply:
> What you describe about matching linux number and zephyr number does not make much sense imho

That's because you're more of lower level networking programmer, while I'm more of higher level networking. So, I appreciate being able to look at it from my PoV, and trusting my word that it's really helpful in our time of pervasive multitasking and switching among multiple systems all the time. Thanks!